components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3215,37 +3219,32 @@ where
             .mut_store()
             .maybe_gc_cache(alive_cache_idx, applied_idx);
         let first_idx = self.fsm.peer.get_store().first_index();
-        let mut compact_idx;
-        if applied_idx > first_idx
-            && applied_idx - first_idx >= self.ctx.cfg.raft_log_gc_count_limit
-        {
-            compact_idx = applied_idx;
-        } else if self.fsm.peer.raft_log_size_hint >= self.ctx.cfg.raft_log_gc_size_limit.0 {
-            compact_idx = applied_idx;
-        } else if replicated_idx < first_idx
-            || replicated_idx - first_idx <= self.ctx.cfg.raft_log_gc_threshold
-        {
-            return;
-        } else {
-            compact_idx = replicated_idx;
-        }
 
-        // Have no idea why subtract 1 here, but original code did this by magic.
-        assert!(compact_idx > 0);
-        compact_idx -= 1;
-        if compact_idx < first_idx {
-            // In case compact_idx == first_idx before subtraction.
+        if !force_compact
+            && (replicated_idx < first_idx
+                || replicated_idx - first_idx <= self.ctx.cfg.raft_log_gc_threshold)
```

Comment:
@BusyJay please take a look here, Can `raft_log_gc_threshold` be reduced to `16` or a less value?